ld/scripttempl/pep.sc

```diff
@@ -41,10 +41,12 @@ if test "${RELOCATING}"; then
     KEEP (SORT(*)(.idata$6))
     KEEP (SORT(*)(.idata$7))'
   R_DIDAT234='
+    __DELAY_IMPORT_DIRECTORY_start__ = .;
     KEEP (SORT(*)(.didat$2))
     KEEP (SORT(*)(.didat$3))
+    __DELAY_IMPORT_DIRECTORY_end__ = .;
     /* These zeroes mark the end of the import list.  */
-    LONG (0); LONG (0); LONG (0); LONG (0); LONG (0);
+    . += (__DELAY_IMPORT_DIRECTORY_end__ - __DELAY_IMPORT_DIRECTORY_start__) ? 8*4 : 0;
```

Comment:
[Microsoft documentation](https://learn.microsoft.com/en-us/windows/win32/debug/pe-format#the-idata-section) says the Directory Table (`.idata$2`) is terminated by a null Directory Entry, which is 20 bytes (`_sizeof(IMAGE_IMPORT_DESCRIPTOR)_`) for both PE and PE+, but I think you can just hard-code `20` as it can't change.

Reply:
yes, that would explain the 5 longs that were copied from the `.idata` section.  But this is [the Delay-Load Directory Table](https://learn.microsoft.com/en-us/windows/win32/debug/pe-format#the-delay-load-directory-table), which is 32 bytes (8 4-byte entries, where the 8*4 came from), and I wanted to avoid filling in a bunch of zeros if there were no delay import directory entries so I did that `+=` trick so I could use a ternary to avoid adding space if there is nothing between the start and end symbols.

Reply:
Ahh I see. Makes sense. Does filling zeroes require `FILL(0)`? The `.idata` section apparently does not have a default fill pattern, and I am uncertain whether the gap is filled with zeroes.

Reply:
I was concerned about that too, but I didn't see any way to restore the `FILL` to its original state after setting `FILL(0)`, and when I tested on i686 at least `.rdata` was filling with zeros already.